staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -217,6 +217,9 @@ func (u *udsGRPCConnector) connect(_ context.Context) (proxier, error) {
 	// See https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/357.
 	tunnelCtx := context.TODO()
 	tunnel, err := client.CreateSingleUseGrpcTunnel(tunnelCtx, udsName, dialOption,
+		grpc.WithBlock(),
+		grpc.WithReturnConnectionError(),
+		grpc.WithTimeout(10*time.Second),
```

Comment:
curious where the 10 second hardcoded timeout is coming from. if we're not using the propagated context, is there a reason we're defaulting to something shorter than the standard http.DefaultTransport dial timeout?

Reply:
10s is arbitrary, but this is kube-apiserver talking over UDS to a local konnectivity-server, so it seems reasonable to expect quite fast.

Reply:
No objections, done.